src/middlewared/middlewared/plugins/pool_/import_pool.py

```diff
@@ -204,117 +204,172 @@ async def import_pool(self, job, data):
     @private
     @job()
     def import_on_boot(self, job):
-        cachedir = os.path.dirname(ZPOOL_CACHE_FILE)
-        if not os.path.exists(cachedir):
-            os.mkdir(cachedir)
-
         if self.middleware.call_sync('failover.licensed'):
+            # HA systems pools are imported using the failover
+            # event logic
             return
 
-        zpool_cache_saved = f'{ZPOOL_CACHE_FILE}.saved'
-        if os.path.exists(ZPOOL_KILLCACHE):
-            with contextlib.suppress(Exception):
-                os.unlink(ZPOOL_CACHE_FILE)
-            with contextlib.suppress(Exception):
-                os.unlink(zpool_cache_saved)
-        else:
-            with open(ZPOOL_KILLCACHE, 'w') as f:
-                os.fsync(f)
-
+        set_cachefile_property = True
         try:
-            stat = os.stat(ZPOOL_CACHE_FILE)
-            if stat.st_size > 0:
-                copy = False
-                if not os.path.exists(zpool_cache_saved):
-                    copy = True
-                else:
-                    statsaved = os.stat(zpool_cache_saved)
-                    if stat.st_mtime > statsaved.st_mtime:
-                        copy = True
-                if copy:
-                    shutil.copy(ZPOOL_CACHE_FILE, zpool_cache_saved)
-        except FileNotFoundError:
-            pass
-
-        job.set_progress(0, 'Beginning pools import')
-
-        pools = self.middleware.call_sync('pool.query', [('status', '=', 'OFFLINE')])
-        for i, pool in enumerate(pools):
-            # Importing pools is currently 80% of the job because we may still need
-            # to set ACL mode for windows
-            job.set_progress(int((i + 1) / len(pools) * 80), f'Importing {pool["name"]}')
-            imported = False
-            if pool['guid']:
-                try:
-                    self.middleware.call_sync('zfs.pool.import_pool', pool['guid'], {
-                        'altroot': '/mnt',
-                        'cachefile': 'none',
-                    }, True, zpool_cache_saved if os.path.exists(zpool_cache_saved) else None)
-                except Exception:
-                    # Importing a pool may fail because of out of date guid database entry
-                    # or because bad cachefile. Try again using the pool name and wihout
-                    # the cachefile
-                    self.logger.error('Failed to import %s', pool['name'], exc_info=True)
-                else:
-                    imported = True
-            if not imported:
-                try:
-                    self.middleware.call_sync('zfs.pool.import_pool', pool['name'], {
-                        'altroot': '/mnt',
-                        'cachefile': 'none',
-                    })
-                except Exception:
-                    self.logger.error('Failed to import %s', pool['name'], exc_info=True)
-                    continue
-
+            self.logger.debug('Creating path components for %r', ZPOOL_CACHE_FILE)
```

Comment:
Are there any statements in this `try` block that could cause an exception?

Reply:
Which path components are we referring to btw here when we talk about their creation ?

Reply:
> Which path components are we referring to btw here when we talk about their creation ?

That's pretty obvious if you look at the `ZPOOL_CACHE_FILE` value 😄

Reply:
> Are there any statements in this `try` block that could cause an exception?

good catch....have no idea how the `os.makedirs` call got lost in my commit 🤷